diff -u -r a/include/fix.h b/include/fix.h
--- a/include/fix.h	2005-07-30 21:02:18.000000000 +0200
+++ b/include/fix.h	2012-02-20 10:52:57.506469215 +0100
@@ -1,6 +1,8 @@
 #ifndef FIX_H
 #define FIX_H
 
+#include <wchar.h>
+
 /* so just #define FIX_NAMSPACE_SYMBOL abcd_ (or whatever namespace
  * you are taking, then everything will just work) */
 #define FIX_XSTR2(x, y) x ## y
diff -u -r a/include/main_system.h b/include/main_system.h
--- a/include/main_system.h	2004-10-25 04:30:36.000000000 +0200
+++ b/include/main_system.h	2012-02-20 10:52:57.506469215 +0100
@@ -33,6 +33,7 @@
 #include <dirent.h>
 #include <signal.h>
 #include <sys/types.h>
+#include <sys/prctl.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
